src/Antiforgery/src/Internal/DefaultAntiforgeryTokenStore.cs

```diff
@@ -57,8 +58,17 @@ public async Task<AntiforgeryTokenSet> GetRequestTokensAsync(HttpContext httpCon
             {
                 // Check the content-type before accessing the form collection to make sure
                 // we report errors gracefully.
-                var form = await httpContext.Request.ReadFormAsync();
-                requestToken = form[_options.FormFieldName];
+                try
+                {
+                    var form = await httpContext.Request.ReadFormAsync();
+                    requestToken = form[_options.FormFieldName];
+                }
+                catch (IOException ex)
+                {
+                    // Reading the request body (which happens as part of ReadFromAsync) may throw an exception if a client disconnects.
+                    // Rethrow this as an AntiforgeryException and allow the caller to handle it as just another antiforgery failure.
```

Comment:
Suggest adding comments like this in the `IValueProviderFactory` implementation additions too